### PR TITLE
chore: Update docstrings for offline stores and retrieval jobs

### PR DESCRIPTION
**Pull request by felixwang9817**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/infra/offline_stores/offline_store.py [hunk 1]
```diff
@@ -152,36 +140,49 @@ def to_arrow(
 
         return pyarrow.Table.from_pandas(features_df)
 
+    @abstractmethod
+    def _to_df_internal(self) -> pd.DataFrame:
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/offline_store.py:144]: The docs should clarify why this exists and is different form `to_df`.

File: sdk/python/feast/infra/offline_stores/offline_store.py [hunk 2]
```diff
@@ -152,36 +140,49 @@ def to_arrow(
...
+        """Synchronously executes the underlying query and returns the result as a pandas dataframe."""
+        pass
+
+    @abstractmethod
+    def _to_arrow_internal(self) -> pyarrow.Table:
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/offline_store.py:149]: same

File: sdk/python/feast/infra/offline_stores/bigquery.py [hunk 3]
```diff
@@ -426,29 +426,27 @@ def _to_df_internal(self) -> pd.DataFrame:
...
-        """
+        """Returns the underlying SQL query."""
         with self._query_generator() as query:
             return query
 
     def to_bigquery(
         self,
-        job_config: bigquery.QueryJobConfig = None,
+        job_config: Optional[bigquery.QueryJobConfig] = None,
         timeout: int = 1800,
```
[the remark this answers is not in the document]
Comment (felixwang9817) [reply sdk/python/feast/infra/offline_stores/bigquery.py:436]: agreed but this is public facing so we'd have to issue some kind of a deprecation warning?

File: sdk/python/feast/infra/offline_stores/offline_store.py (hunk 1 above)
Comment (felixwang9817) [reply sdk/python/feast/infra/offline_stores/offline_store.py:144]: good catch

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**